python/cudf/cudf/core/udf/strings_lowering.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import operator
+
+from numba import types
+from numba.core import cgutils
+from numba.cuda.cudaimpl import lower as cuda_lower
+
+from strings_udf._typing import size_type, string_view
+from strings_udf.lowering import (
+    string_view_contains_impl,
+    string_view_count_impl,
+    string_view_endswith_impl,
+    string_view_find_impl,
+    string_view_isalnum_impl,
+    string_view_isalpha_impl,
+    string_view_isdecimal_impl,
+    string_view_isdigit_impl,
+    string_view_islower_impl,
+    string_view_isspace_impl,
+    string_view_isupper_impl,
+    string_view_len_impl,
+    string_view_rfind_impl,
+    string_view_startswith_impl,
+)
+
+from cudf.core.udf.masked_typing import MaskedType
+
+
+@cuda_lower(len, MaskedType(string_view))
```

Comment:
See my suggestions in `strings_udf/.../lowering.py` around using decorators/helper functions for dynamic generation.

Reply:
Marking this as resolved with the refactor in https://github.com/rapidsai/cudf/pull/11319/commits/8938f1b648f9e5caed94cbf5b2532a746d18c5c9